### PR TITLE
Added error functionality while addition of admin as moderator to an instance

### PR DESCRIPTION
**Pull request by singhsanket143**
The pull request is implementation for the feature listed in issue #15448. Earlier Admin of an instance can be added as a moderator too which was erroneous. Now, this issue is resolved. Now one cannot add admin of an instance as a moderator.
image

### CODE REVIEW

File: client/views/add/add.js [hunk 1]
```diff
@@ -92,13 +92,17 @@ Template.add.events({
...
       // If the result is an object, there was an error
-      if (typeof result === 'object') {
+      if (typeof result === 'object' && result.length > 0) {
         // Alert the error
         for (let i = 0; i < result.length; i++) {
           if (result[i].name === 'moderators') {
             showModsError('You can only assign 4 moderators per instance.');
             return false;
           }
+          if(result[i].name === 'owner') { // Check is the server returned error corresponding to the addition of owner as moderator
```
Comment (peter-hank) [inline client/views/add/add.js]: @singhsanket143 move the comments up or down so they start at the beginning of the line

File: client/views/add/add.js [hunk 2]
```diff
@@ -92,13 +92,17 @@ Template.add.events({
...
+            showModsError('You cannot add owner of the instance as moderator'); // Display the error message
```
Comment (peter-hank) [inline client/views/add/add.js]: `You cannot add owner of the instance as moderator` > `You cannot add the owner of the instance as a moderator`

File: server/methods.js [hunk 3]
```diff
@@ -270,7 +270,13 @@ Meteor.methods({
       const instance = Instances.findOne({
         _id: instanceid,
       });
+      var found = mods.find(function(element){
```
Comment (peter-hank) [inline server/methods.js]: `var found = mods.find(function(element){` > `var found = mods.find(function(element) {`

File: server/methods.js [hunk 4]
```diff
@@ -270,7 +270,13 @@ Meteor.methods({
...
+        return element === instance.admin;
+      });
       if (email === instance.admin) {
+        if(typeof found !== 'undefined') {
+            return [{name: 'owner', type: 'Invalid candidate for moderator', value: instance.admin}];
```
Comment (peter-hank) [inline server/methods.js]: `Invalid candidate for moderator` > `An invalid candidate for a moderator`

**Review by peter-hank — CHANGES_REQUESTED** [review]
Ok, we need some updates.

File: client/views/add/add.js (hunk 1 above)
Comment (singhsanket143) [reply client/views/add/add.js]: I have rectified the issue for comments in the latest commits

File: client/views/add/add.js (hunk 2 above)
Comment (singhsanket143) [reply client/views/add/add.js]: Updated the error message

File: server/methods.js (hunk 3 above)
Comment (singhsanket143) [reply server/methods.js]: updated indentation

File: server/methods.js (hunk 4 above)
Comment (singhsanket143) [reply server/methods.js]: updated

File: client/views/add/add.js [hunk 5]
```diff
@@ -92,13 +92,18 @@ Template.add.events({
...
+      if (typeof result === 'object' && result.length > 0) {
         // Alert the error
         for (let i = 0; i < result.length; i++) {
           if (result[i].name === 'moderators') {
             showModsError('You can only assign 4 moderators per instance.');
             return false;
           }
+          // Check is the server returned error corresponding to the addition of owner as moderator
+          if(result[i].name === 'owner') {
+            showModsError(`${result[i].value} is already an owner of the instance and has the privileges of a moderator.`); // Display the error message
```
Comment (peter-hank) [inline client/views/add/add.js]: @singhsanket143 can we move the comment up?

Comment (singhsanket143) [reply client/views/add/add.js]: @peter-hank I have Updated the comments

**Pull request merged**